server/notifications.model.js

```diff
@@ -0,0 +1,66 @@
+const mongoose = require('mongoose');
+const _ = require('lodash');
+
+const NotificationSchema = new mongoose.Schema({
+	dismissKey : { type: String, unique: true, required: true },
+	title      : { type: String, default: '' },
+	text       : { type: String, default: '' },
+	createdAt  : { type: Date, default: Date.now },
+	startAt    : { type: Date, default: Date.now },
+	stopAt     : { type: Date, default: Date.now },
+}, { versionKey: false });
+
+NotificationSchema.statics.addNotification = async function(data) {
+	if(!data.dismissKey) return { success: false, message: 'Dismiss key is required!' };
+
+	const defaults = {
+		title   : '',
+		text    : '',
+		startAt : new Date(),
+		stopAt  : new Date(),
+	};
+
+	const notificationData = _.defaults(data, defaults);
+
+	if(!(notificationData.startAt instanceof Date) || !(notificationData.stopAt instanceof Date)) {
+		return { success: false, message: 'Invalid date format for startAt or stopAt' };
```

Comment:
I see you added some more tests. Good!

I wanted to check what the purpose of is extra `success` field is, since the only place it shows up is in the tests. Are you still working on something where it will be used elsewhere?  If it's only going to ever appear in the tests we can just remove it.